app/src/main/java/org/mozilla/focus/session/SessionManager.java

```diff
@@ -282,6 +293,28 @@ private void addSession(Session session) {
         }
     }
 
+    private void addNewTabSession(final Session session, Context context) {
+        if (currentSessionUUID == null || Settings.getInstance(context).shouldOpenNewTabs()) {
+            currentSessionUUID = session.getUUID();
+        } else {
+            // Show Snackbar that allow users to switch to tab they just opened
+            final Snackbar snackbar = ViewUtils.getBrandedSnackbar(
```

Comment:
Shouldn't this UI logic be in some UI logic related class, `SessionManager` simply deals with sessions and the currently opened sessions. Maybe in a fragment or `MainActivity`?